mindsdb_native/libs/backends/lightwood.py

```diff
@@ -395,37 +392,54 @@ def predict(self, mode='predict', ignore_columns=None, all_mixers=False):
         else:
             run_df = df
 
-        predictions = self.predictor.predict(when_data=run_df)
+        if self.transaction.lmd['tss']['is_timeseries']:
+            run_df, _ = self._create_timeseries_df(run_df)
+
+        predictions = {}
+        predictions[0] = self.predictor.predict(when_data=run_df)
+
+        # additional predictions for timeseries
+        timesteps = self.transaction.lmd.get('nr_predictions', 1)
+        if self.transaction.lmd['tss']['is_timeseries']:
+            for t in range(1, timesteps):
+                displaced_df = copy.deepcopy(df)
+                for target in self.predictor._output_columns:
+                    displaced_df[target] = predictions[t-1][target]['predictions']
+                # if self.transaction.lmd['tss']['use_previous_target']:
+                displaced_df, _ = self._create_timeseries_df(displaced_df)
+                predictions[t] = self.predictor.predict(when_data=displaced_df)
 
         formated_predictions = {}
-        for k in predictions:
-            formated_predictions[k] = predictions[k]['predictions']
-
-            model_confidence_dict = {}
-            for confidence_name in ['selfaware_confidences','loss_confidences', 'quantile_confidences']:
-
-                if confidence_name in predictions[k]:
-                    if k not in model_confidence_dict:
-                        model_confidence_dict[k] = []
-
-                    for i in range(len(predictions[k][confidence_name])):
-                        if len(model_confidence_dict[k]) <= i:
-                            model_confidence_dict[k].append([])
-                        conf = predictions[k][confidence_name][i]
-                        # @TODO We should make sure lightwood never returns confidences above or bellow 0 and 1
-                        if conf < 0:
-                            conf = 0
-                        if conf > 1:
-                            conf = 1
-                        model_confidence_dict[k][i].append(conf)
-
-            for k in model_confidence_dict:
-                model_confidence_dict[k] = [np.mean(x) for x in model_confidence_dict[k]]
-
-            for k  in model_confidence_dict:
-                formated_predictions[f'{k}_model_confidence'] = model_confidence_dict[k]
-
-            if 'confidence_range' in predictions[k]:
-                formated_predictions[f'{k}_confidence_range'] = predictions[k]['confidence_range']
+        for t in range(timesteps):
+            formated_predictions[t] = {}
```

Comment:
I don't quite understand the logic here, it might be better to:

1) have predictions == predictions[-1] for the purpose of computing the confidence (do an if timesteps > 1 and set it somewhere)
2) have formated_predictions[k] be an array of arrays [[Y1_T+1,Y1_T+2,Y1_T+3], [Y2_T+1,Y2_T+2,Y2_T+3], [Y3_T+1,Y3_T+2,Y3_T+3]] ... etc